functions/Get-DbaAgentJobOutputFile.ps1

```diff
@@ -0,0 +1,144 @@
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true, 
+                   ValueFromRemainingArguments=$false, 
+                   Position=1)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory=$false,HelpMessage='Want to open the file')] 
+        [switch]$OpenFile)
+    DynamicParam {
+            # Set the dynamic parameters' name
+            $ParameterName = 'JobName'
+            
+            # Create the dictionary 
+            $RuntimeParameterDictionary = New-Object -TypeName System.Management.Automation.RuntimeDefinedParameterDictionary
+
+            # Create the collection of attributes
+            $AttributeCollection = New-Object -TypeName System.Collections.ObjectModel.Collection[System.Attribute]
+            
+            # Create and set the parameters' attributes
+            $ParameterAttribute = New-Object -TypeName System.Management.Automation.ParameterAttribute
+            $ParameterAttribute.Mandatory = $true
+            $ParameterAttribute.Position = 1
+
+            # Add the attributes to the attributes collection
+            $AttributeCollection.Add($ParameterAttribute)
+
+            # Generate and set the ValidateSet 
+            $Server = Connect-DbaSqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+            $arrSet = ($server.JobServer.Jobs).Name
+            $ValidateSetAttribute = New-Object -TypeName System.Management.Automation.ValidateSetAttribute -ArgumentList ($arrSet)
+
+            # Add the ValidateSet to the attributes collection
+            $AttributeCollection.Add($ValidateSetAttribute)
+
+            # Create and return the dynamic parameter
+            $RuntimeParameter = New-Object -TypeName System.Management.Automation.RuntimeDefinedParameter -ArgumentList ($ParameterName, [string], $AttributeCollection)
+            $RuntimeParameterDictionary.Add($ParameterName, $RuntimeParameter)
+            return $RuntimeParameterDictionary
+    }
+
+    begin {
+        # Bind the parameter to a friendly variable
+        $JobName = $PsBoundParameters[$ParameterName]
+        $Server = Connect-DbaSqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+    }
+    process
+    {
+    $Job = $server.JobServer.Jobs[$JobName]
+
+        $Servername = $Server.ComputerNamePhysicalNetBIOS
+
+    foreach($Step in $Job.JobSteps)
+    {
+        $fileName = $Step.OutputFileName
+
+        if($fileName -eq '')
+        {
+            Write-Output "$($step.Name) has no output file"
+        }
+        else
+        {
+            
+            if($FileName.StartsWith('\\') -eq $false)
+            {
+                $fileName = '\\' + $Servername + '\' + $Filename.Replace(':','$')
+                Write-Output "$($step.Name) - $fileName"
+            }
+            else
+            {
+                Write-Output "$($step.Name) - $fileName "
+            }
+
+            if($OpenFile)
```

Comment:
@PowerDBAKlaas  - where do you think we'll go with this? Ive seen similar "helpers" in SqlServer. This could be just piped to Invoke-Item.